compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1548,9 +1550,33 @@ impl<'a> Parser<'a> {
             Ok(self.mk_expr_err(lo))
         } else {
             let msg = "expected `while`, `for`, `loop` or `{` after a label";
-            self.struct_span_err(self.token.span, msg).span_label(self.token.span, msg).emit();
+
+            let mut err = self.struct_span_err(self.token.span, msg);
+            err.span_label(self.token.span, msg);
+
             // Continue as an expression in an effort to recover on `'label: non_block_expr`.
-            self.parse_expr()
+            let expr = self.parse_expr().map(|expr| {
+                let span = expr.span;
+                let sugg_msg = "consider enclosing expression in a block";
+                let suggestions = vec![
+                    (span.shrink_to_lo(), "{".to_owned()),
+                    (span.shrink_to_hi(), "}".to_owned()),
+                ];
+
+                err.multipart_suggestion_verbose(
+                    sugg_msg,
+                    suggestions,
+                    Applicability::MachineApplicable,
+                );
+
+                // Replace `'label: non_block_expr` with `'label: {non_block_expr}` in order to supress future errors about `break 'label`.
```

Comment:
Should we gate this on the `label_break_value` feature being enabled? I think you can get the feature from `sess.features_untracked().label_break_value`.

If not, then could we walk the AST to look for usages of `label`? If we don't see any, then I would prefer a suggestion to remove the label.

Reply:
Oh wait... `ParserSess` doesn't have access to the features. Ugh. Maybe we could only suggest this on nightly? Maybe that's actually kinda useless.

Reply:
Well, I was wandering the same and didn't come to any conclusions. Walking the AST may be a good idea, it should provide better diagnostics, I think.

I didn't do `is_nightly` check mostly because the feature is headed towards stabilization anyway.

Reply:
Ok, let's just walk the AST if that's not too difficult, and not restrict to only nightly.

Reply:
Maybe we should also suggest removing the label? 🤔

Reply:
This was actually easier than I thought :P